### ISSUE TITLE
Create a logging scheme

### ISSUE THREAD

**Issue by luethan2025** [luethan2025/18794#1]
#### Add a logger to log all parameters/hyperparameters as a part of the training loop
```python
# trainer.py
def train(model, optimizer, lr_scheduler, logger, criterion, trainloader, testloader, epochs, device):
    train_losses = []
    test_losses = []

    model.to(device)
    for epoch in range(epochs):
        running_loss = 0.0
        model.train()

        for inputs, labels in trainloader:
            optimizer.zero_grad()
            inputs, labels = inputs.to(device), labels.to(device)
            outputs = model(inputs)
            loss = criterion(outputs, labels)
            loss.backward()
            optimizer.step()
            running_loss += loss.item()
        lr_scheduler.step()

        train_loss = running_loss / len(trainloader)
        train_losses.append(train_loss)

        test_loss = evaluate_loss(model, criterion, testloader, device)
        test_losses.append(test_loss)

        test_accuracy = evaluate_accuracy(model, testloader, device)

        #TODO
        logger.log()

    return train_losses, test_losses
```

#### Current parameters of interest
- Total loss
- Total accuracy
- Recall
- Precision
- F1-score
- Intersection over Union (IoU)
- Current learning rate
